src/Views/AppInfoView.vala

```diff
@@ -323,30 +339,39 @@ namespace AppCenter.Views {
                 } catch (Error e) {
                     debug (e.message);
                     // The file is likely to not being found.
-                    screenshot_stack.visible_child = app_screenshot_not_found;
-                    return;
+                    return screenshot_load_error_label ();
                 }
 
                 GLib.FileUtils.close (fd);
             } else {
                 critical ("Error create the temporary file: GFileError #%d", GLib.FileUtils.error_from_errno (GLib.errno));
                 fileimage = File.new_for_uri (url);
                 if (fileimage.query_exists () == false) {
-                    screenshot_stack.visible_child = app_screenshot_not_found;
-                    return;
+                    return screenshot_load_error_label ();
                 }
             }
 
-            Idle.add (() => {
-                try {
-                    app_screenshot.pixbuf = new Gdk.Pixbuf.from_file_at_scale (fileimage.get_path (), 800, 600, true);
-                    screenshot_stack.visible_child = app_screenshot;
-                } catch (Error e) {
-                    critical (e.message);
-                }
+            try {
+                var image = new Gtk.Image ();
+                image.width_request = 800;
+                image.height_request = 500;
+                image.icon_name = "image-x-generic";
+                image.halign = Gtk.Align.CENTER;
+                image.pixbuf = new Gdk.Pixbuf.from_file_at_scale (fileimage.get_path (), 800, 600, true);
+
+                return image;
+            } catch (Error e) {
+                critical (e.message);
+                return screenshot_load_error_label ();
+            }
+        }
 
-                return GLib.Source.REMOVE;
-            });
+        private Gtk.Label screenshot_load_error_label () {
```

Comment:
This should only run once if no screenshot was able to be loaded :) That way you don't end up with a stack of images and errors, or even yet, a stack full of errors. Remember to do it on an Idle block